mne/channels/montage.py

```diff
@@ -791,8 +826,20 @@ def dig_ch_pos(self):
         return self._ch_pos()
 
     def _get_ch_pos(self):
-        return dict(zip(self.ch_names,
-                        _foo_get_data_from_dig(self.dig).dig_ch_pos_location))
+        pos = [d['r'] for d in _get_dig_eeg(self.dig)]
+        return dict(zip(self.ch_names, pos))
+
+    def _get_dig_point_name(self):
+        NAMED_KIND = (FIFF.FIFFV_POINT_EEG, FIFF.FIFFV_EEG_CH)
+        _ch_names = iter(self.ch_names)
+
+        # XXX: StopIteration is deprecated since py3.5
+        # see https://stackoverflow.com/questions/43617399/how-to-get-rid-of-warning-deprecationwarning-generator-ngrams-raised-stopiter  # noqa
+        try:
+            for d in self.dig:
+                yield next(_ch_names) if d['kind'] in NAMED_KIND else None
+        except (StopIteration):
+            return
```

Comment:
> The idea is that I would like to get a list of either the ith ch_name or None.

This solution above is a challenge to parse. I find something like this more readable, dealing with indexing and vectorization rather than fighting with iterator semantics:
```
is_eeg = np.array([d['kind'] in NAMED_KIND for d in self.dig])  # XXX FIFFV_EEG_CH is not a valid DigPoint type...
if len(self.ch_names) < is_eeg.sum():
    raise RuntimeError(...)
dig_names = [None] * len(self.dig)
for ch_name_idx, dig_idx in enumerate(np.where(is_eeg)[0]):
    dig_names[dig_idx] = self.ch_names[ch_name_idx]
```

Reply:
yes, this `try` thing makes it way less nice. Otherwise, it was quite compact.

regarding the channels. `FIFF_EEG_CH` puzzles me it is set in the old montage in 'chs'. That's why I added it but it lead to a rabbit whole. So what is the difference? is it that `FIFF_EEG_CH` is the same as `FIFFV_POINT_EEG` but one is for `info['chs']..` and the other one is for `info['dig']`?